site/content/docs/02-template-syntax.md

```diff
@@ -1289,6 +1289,27 @@ Named slots allow consumers to target specific areas. They can also have fallbac
 </div>
 ```
 
+---
+
+`$$slots` is an object whose keys are the names of the slots passed into the component by the parent. If the parent leaves a named slot empty, `$$slots` will not have a key for that slot. This allows components to render a slot (and other elements, like wrappers for styling) only if the parent provides children for it.
```

Comment:
This isn't quite true. If a slot is explicitly passed with no content, `$$slots.foo` will still be present. This is basically what #5312 is asking for, and I'm not sure how that's going to be handled yet.

Reply:
Good point. I updated to make the difference between omitted slots and empty slots more clear. My language feels a little strained in this sentence:

> If the parent does not pass in a slot with a particular name, that name will not be a present in `$$slots`.

I was trying to avoid saying "If the parent does not pass in a named slot," which sounds ambiguous to my ear -- could be misinterpreted as "If the parent does not pass in _any_ named slots." But I'm not sure if "a slot with a particular name" is the best way to describe what we mean, either. Feedback appreciated.

I also added a note under the `$$slots` code sample to hammer home the behavior for explicitly empty slots, which may or may not be necessary -- feedback also appreciated on that.